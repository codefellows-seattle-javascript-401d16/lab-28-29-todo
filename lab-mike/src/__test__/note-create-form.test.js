import React from 'react';
import {shallow, mount} from 'enzyme';
import NoteForm from '../component/note-form';

describe('testing NoteForm', () => {
  test('should have correct default state', () => {
    let wrapper = shallow(<NoteForm handleSubmit={() => {}} />);
    expect(wrapper.state('inputText')).toBe('');
  });
  test('note form with props', () => {
    let mockNote = {
      inputText: 'potato',
    };
    let wrapper = shallow(<NoteForm note={mockNote} />);
    expect(wrapper.state()).toEqual(mockNote);
    wrapper.find('button').simulate('click');
  });
  test('testing inputText change onChange handler', () => {
    let wrapper = shallow(<NoteForm handleSubmit={() => {}} />);
    wrapper.find('input').simulate('change', {target: {value: '123'}});
    expect(wrapper).toMatchSnapshot();
  });
  test('additional input change to change state', () => {
    let wrapper = mount(<NoteForm />);
    wrapper.find('input[name="inputText"]').simulate('change', {
      target: {
        name: 'inputText',
        value: '456',
      },
    });
    expect(wrapper.state('inputText')).toEqual('456');
  });
});
