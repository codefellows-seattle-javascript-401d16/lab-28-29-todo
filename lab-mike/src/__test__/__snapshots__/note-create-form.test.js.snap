// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing NoteCreateForm testing inputText change onChange handler 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    onSubmit={[Function]}
>
    <input
        name="inputText"
        onChange={[Function]}
        type="text"
        value={undefined}
    />
    <button
        type="submit"
    >
        Submit Text
    </button>
</form>,
  "nodes": Array [
    <form
      onSubmit={[Function]}
>
      <input
            name="inputText"
            onChange={[Function]}
            type="text"
            value={undefined}
      />
      <button
            type="submit"
      >
            Submit Text
      </button>
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <NoteCreateForm
        handleNoteCreate={[Function]}
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": NoteCreateForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "handleSubmit": [Function],
        "props": Object {
          "handleNoteCreate": [Function],
        },
        "refs": Object {},
        "state": Object {
          "inputText": "123",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          onSubmit={[Function]}
>
          <input
                    name="inputText"
                    onChange={[Function]}
                    type="text"
                    value={undefined}
          />
          <button
                    type="submit"
          >
                    Submit Text
          </button>
</form>,
        "_debugID": 6,
        "_renderedOutput": <form
          onSubmit={[Function]}
>
          <input
                    name="inputText"
                    onChange={[Function]}
                    type="text"
                    value={undefined}
          />
          <button
                    type="submit"
          >
                    Submit Text
          </button>
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NoteCreateForm
    handleNoteCreate={[Function]}
/>,
}
`;
